src/include/function/gds/gds.h

```diff
@@ -17,13 +18,23 @@ namespace function {
 
 // Struct maintaining GDS specific information that needs to be obtained at compile time.
 struct GDSBindData {
+    std::shared_ptr<binder::Expression> nodeOutput;
+    // If outputAsNode is true, we will scan all properties of the node.
+    // Otherwise, we return node ID only.
+    // This can be extended by allowing user specifify what properties to return in the arguments.
```

Comment:
Remove this "This can be extended by allowing user specifify what properties to return in the arguments." I think we should make it an item in an issue called "Algorithms Package TODOs".